tests/unit/test_ut_scenario_loader.py

```diff
@@ -298,3 +303,130 @@ def test_scenarios_no_such_target(self):
                 class MyTests(ScenarioTestsBase):
                     pass
                 MyTests()
+
+    def test_scenarios_check_mappings(self):
+        """Check for all YAML tests and scenarios to determine whether
+        every scenario has at least one test.
+        """
+
+        tests_root_path = os.path.join(utils.DEFS_TESTS_DIR, 'scenarios')
+        scenarios_root_path = os.path.join(utils.DEFS_DIR, 'scenarios')
+
+        # This list contains the full paths to all scenario test cases.
+        # This information is used for reporting the number of avaliable
+        # test cases.
+        all_tests = []
+
+        # A collection of all tests and their respective scenarios. Scenario
+        # name is used as a key, where the value is list of tests associated
+        # with the scenario.
+        test_scenario_mappings = {}
+
+        # Iterate over all subdirectories of `hotsos/defs/tests` and try to
```

Comment:
The YDefsLoader class [1] exists for the purpose of loading all definitions in the scenario and event modules. Could we perhaps leverage it here instead of re-implementing the logic? We would also have the added benefit of extra test cycles on that code.

[1] https://github.com/canonical/hotsos/blob/10b04e1a59ec782c478f6fefe7bf69fb586bde8d/hotsos/core/ycheck/engine/common.py#L9

Reply:
I tried using it today but it further complicated things, maybe I need a different approach. I'll give it another shot tomorrow with a  fresh mind.

Reply:
I couldn't find a simple solution using YDefsLoader. The problem is YDefsLoader is giving me the YAML contents in a parsed tree, but what I really need is the path to the actual YAML files, which I couldn't find in the tree itself. I guess I can find the path to the files by joining node names to form a path, but AFAICT it'll not be simpler nor prettier than the status quo.